tests/common.rs

```diff
@@ -90,3 +94,42 @@ impl BlobCache for NoCacher {
 
     fn insert(&self, _: ValueLogId, _: &ValueHandle, _: UserValue) {}
 }
+
+impl FDCache for NoCacher {
+    fn get(&self, _: ValueLogId, _: BlobFileId) -> Option<BufReader<File>> {
+        None
+    }
+    fn insert(&self, _: ValueLogId, _: BlobFileId, _: BufReader<File>) {}
+}
+
+#[derive(Clone, Default)]
+pub struct InMemCacher {
+    fd_cache: Arc<Mutex<HashMap<(ValueLogId, BlobFileId), File>>>,
+    fd_hit_count: Arc<RefCell<u32>>,
+}
+impl InMemCacher {
+    pub(crate) fn get_hit_count(&self) -> u32 {
+        *self.fd_hit_count.borrow()
+    }
+}
+
+impl FDCache for InMemCacher {
+    fn get(&self, vlog_id: ValueLogId, blob_file_id: BlobFileId) -> Option<BufReader<File>> {
+        let lock = self.fd_cache.lock().unwrap();
+        let fd = match lock.get(&(vlog_id, blob_file_id)) {
+            Some(fd) => fd,
+            None => return None,
+        };
+
+        let fd_clone = fd.try_clone().unwrap();
+        *self.fd_hit_count.borrow_mut() += 1;
```

Comment:
This is a cache *miss*, because the cache lookup above did not return a cached descriptor. FWIW, we could count both hits and misses.